zeebe/gateway-rest/src/main/java/io/camunda/zeebe/gateway/rest/util/OpenApiYamlLoader.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.zeebe.gateway.rest.util;
+
+import io.swagger.v3.oas.models.OpenAPI;
+import io.swagger.v3.parser.OpenAPIV3Parser;
+import io.swagger.v3.parser.core.models.SwaggerParseResult;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.core.io.ClassPathResource;
+
+/** Utility class for loading OpenAPI specifications from YAML files. */
+public final class OpenApiYamlLoader {
+
+  private static final Logger LOG = LoggerFactory.getLogger(OpenApiYamlLoader.class);
+
+  // Prevent instantiation
+  private OpenApiYamlLoader() {}
+
+  /**
+   * Loads an OpenAPI specification from a YAML file located in the classpath.
+   *
+   * @param yamlPath the path to the YAML file relative to the classpath
+   * @return the parsed OpenAPI specification
+   * @throws OpenApiLoadingException if the YAML file cannot be loaded or parsed
+   */
+  public static OpenAPI loadOpenApiFromYaml(final String yamlPath) {
+    try {
+      final String yamlContent = loadYamlContent(yamlPath);
+      final SwaggerParseResult result = new OpenAPIV3Parser().readContents(yamlContent);
+
+      if (result.getOpenAPI() == null) {
+        final String errorMsg = "Failed to parse OpenAPI YAML: " + yamlPath;
+        if (result.getMessages() != null && !result.getMessages().isEmpty()) {
+          LOG.error("{}, Parse errors: {}", errorMsg, result.getMessages());
+        }
+        throw new OpenApiLoadingException(errorMsg);
```

Comment:
❓ what is the impact of throwing this exception here? Does it prevent the cluster from startup? or just affect the user if the swagger endpoint is actually called?